File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    P = dict()
    for p in participant:
        if p in P: P[p] += 1
        else: P[p] = 1
    for c in completion:
        P[c] -= 1
    ans = ""
    for key, value in P.items():
        if value > 0: ans = key
    return ans

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    
    map<string, int> Cnt;
    for (auto it = completion.begin(); it != completion.end(); ++it)
    {
        if (Cnt.find(*it) != Cnt.end())
        {
            Cnt[*it] += 1;
        }
        else
        {
            Cnt[*it] = 1;
        }
    }
    
    string answer = "";
    for (auto &name : participant)
    {
        if ((Cnt.find(name) != Cnt.end()) && (Cnt[name] != 0)){
            Cnt[name] -= 1;
        }
        else{
            answer += name;
            break;
        }
    }
    
    return answer;
}
            